### PR TITLE
Create bridge nat and change config

### PR DESCRIPTION
**Pull request by qingwusunny**
- change config to add enable proxy (Todo add update logic for config)
- create nat bridge in init-agent script

### CODE REVIEW

File: cmd/everoute-agent/config.go [hunk 1]
```diff
@@ -62,10 +70,10 @@ func getAgentConfig() (*agentConfig, error) {
 	return &agentConfig, nil
 }
 
-func getDatapathConfig() (*datapath.Config, error) {
+func getDatapathConfig() (*datapath.Config, *datapath.AgentConf, error) {
```
Comment (wu-cl) [inline cmd/everoute-agent/config.go:73]: Could we merge two configs ?

File: build/script/init_agent [hunk 2]
```diff
@@ -67,18 +67,22 @@ LOCAL_IP_RULE_PREF=200
 
 DEFAULT_BRIDGE=`cat ${AGENT_CONFIG_PATH} | grep datapathConfig: -A1 | grep -v datapathConfig: | awk -F ':' '{print $1}' | awk '$1=$1'`
 LOCAL_GW_IP=`cat ${AGENT_CONFIG_PATH} | grep localGwIP | awk -F ':' '{print $2}' | awk '$1=$1'`
+ENABLE_PROXY=`cat ${AGENT_CONFIG_PATH} | grep enableProxy | awk -F ':' '{print $2}' | awk '$1=$1'`
```
[the remark this answers is not in the document]
Comment (qingwusunny) [reply build/script/init_agent:70]: record as a issue [#569] (https://github.com/everoute/everoute/issues/569).

File: cmd/everoute-agent/config.go (hunk 1 above)
Comment (qingwusunny) [reply cmd/everoute-agent/config.go:73]: use config to store static config
use info to store the info from apiserver or somewhere

File: pkg/agent/datapath/multiBridgeDatapath.go [hunk 3]
```diff
@@ -599,9 +625,15 @@ func InitializeVDS(datapathManager *DpManager, vdsID string, ovsbrName string, s
 	go datapathManager.BridgeChainMap[vdsID][LOCAL_BRIDGE_KEYWORD].(*LocalBridge).cleanLocalEndpointIPAddrWorker(
 		IPAddressCacheUpdateInterval, IPAddressTimeout, stopChan)
 
-	if err := SetPortNoFlood(datapathManager.BridgeChainMap[vdsID][LOCAL_BRIDGE_KEYWORD].(*LocalBridge).name,
-		int(datapathManager.BridgeChainPortMap[ovsbrName][LocalToPolicySuffix])); err != nil {
-		log.Fatalf("Failed to set local to policy port with no flood port mode, %v", err)
+	for _, portSuffix := range []string{LocalToPolicySuffix, LocalToNatSuffix} {
+		if datapathManager.BridgeChainPortMap[ovsbrName][portSuffix] == 0 {
```
Comment (wu-cl) [inline pkg/agent/datapath/multiBridgeDatapath.go:629]: If disable CNI，datapathManager.BridgeChainPortMap[ovsbrName][LocalToNatSuffix] may be a nil pointer ？

File: pkg/agent/datapath/multiBridgeDatapath.go [hunk 4]
```diff
@@ -599,9 +625,15 @@ func InitializeVDS(datapathManager *DpManager, vdsID string, ovsbrName string, s
...
+			log.Infof("Port %s in local bridge doesn't exist, skip set no flood port mode", portSuffix)
+			continue
+		}
+		if err := SetPortNoFlood(datapathManager.BridgeChainMap[vdsID][LOCAL_BRIDGE_KEYWORD].(*LocalBridge).name,
```
Comment (echkenluo) [inline pkg/agent/datapath/multiBridgeDatapath.go:633]: Are there specific unknown unicast packets that need to be flooded to the nat bridge？

File: pkg/agent/datapath/multiBridgeDatapath.go (hunk 3 above)
Comment (qingwusunny) [reply pkg/agent/datapath/multiBridgeDatapath.go:629]: uint32, the default value is 0, not nil

File: pkg/agent/datapath/multiBridgeDatapath.go [hunk 5]
```diff
@@ -214,9 +213,15 @@ type AgentConf struct {
...
-	InternalIPs   []string          // internal IPs
+type DpManagerConfig struct {
+	ManagedVDSMap map[string]string   // map vds to ovsbr-name
+	InternalIPs   []string            // internal IPs
+	EnableCNI     bool                // enable CNI in Everoute
+	CNIConfig     *DpManagerCNIConfig // config related CNI
+}
+
+type DpManagerCNIConfig struct {
+	EnableProxy bool // enable proxy
```
[the remark this answers is not in the document]
Comment (qingwusunny) [reply pkg/agent/datapath/multiBridgeDatapath.go:224]: yes

File: pkg/agent/datapath/multiBridgeDatapath.go (hunk 4 above)
Comment (qingwusunny) [reply pkg/agent/datapath/multiBridgeDatapath.go:633]: no, only specific packets need to output to nat bridge. And the l3forward in nat bridge will depend to the flow from agent

**Review by wu-cl — APPROVED** [review]

File: pkg/agent/datapath/multiBridgeDatapath.go (hunk 4 above)
Comment (qingwusunny) [reply pkg/agent/datapath/multiBridgeDatapath.go:633]: local bridge need to add replay logic for flow to nat bridge

Comment (echkenluo) [reply pkg/agent/datapath/multiBridgeDatapath.go:633]: We need to replay service traffic redirect flow in local bridge at least

**Review by echkenluo — APPROVED** [review]

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**